src/server/model_run.R

```diff
@@ -13,19 +13,23 @@ modelRun <- function(input, output, spectrumFilesState, surveyAndProgramData) {
         # converting survey data to the expected format
         surveyAsDataTable <- as.data.table(surveyAndProgramData$survey, keep.rownames = TRUE)
 
-        out <- fitModel(surveyAsDataTable, surveyAndProgramData$program(), spectrumFilesState$combinedData())
+        out <- tryCatch({
+            fitModel(surveyAsDataTable, surveyAndProgramData$program(), spectrumFilesState$combinedData())
 
-        # model fit results
-        likdat <- out$likdat
-        fp <- out$fp
-        mod <- eppasm::simmod.specfp(fp)
+            # model fit results
```

Comment:
I'm not sure what the conditions on the data are for the model fitting to work; if we can find them out we could keep the model run tab disabled until they're met, but for now I'm letting the user proceed if they have at least one row of programmatic data and catching errors here.